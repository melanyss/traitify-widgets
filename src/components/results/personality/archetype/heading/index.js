import PropTypes from "prop-types";
import DangerousHTML from "lib/helpers/dangerous-html";
import {useDidMount, useDidUpdate} from "lib/helpers/hooks";
import {dig} from "lib/helpers/object";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

function PersonalityArchetypeHeading(props) {
  const {assessment, deck, followDeck, getOption, isReady, translate, ui} = props;
  const personality = dig(assessment, ["archetype"]);
  const state = {};

  useDidMount(() => { ui.trigger("PersonalityArchetype.initialized", {props, state}); });
  useDidMount(() => { followDeck(); });
  useDidUpdate(() => { ui.trigger("PersonalityArchetype.updated", {props, state}); });

  if(!isReady("results")) { return null; }
  if(!isReady("deck")) { return null; }
  if(!personality) { return null; }

  const disabledComponents = getOption("disabledComponents") || [];
  if(disabledComponents.includes("PersonalityArchetype")) { return null; }

  const badge = personality.details.find(({title}) => title === "Badge");
  const perspective = getOption("perspective");
  const description = (
    perspective === "thirdPerson"
      ? personality.details.find(({title}) => title === "Hiring Manager Description")
      : personality.details.find(({title}) => title === "Candidate Description")
  );

  return (
    <div className={style.container}>
      {perspective === "thirdPerson" ? (
        <div className={style.thirdPersonDetails}>
          <div className={description && style.badgeAndName}>
            {badge && <img alt={personality.name} src={badge.body} />}
            <DangerousHTML
              html={translate("personality_heading_third_person", {
                deck_name: deck.name,
                personality: `<br /><span>${personality.name}</span><br />`
              })}
              className={style.personalityHeading}
              tag="h2"
            />
          </div>
          {description && <span className={style.divider} />}
          {description && <span className={style.body}>{description.body}</span>}
        </div>
      ) : [
        <div key="heading" className={style.details}>
          {badge && <img alt={personality.name} src={badge.body} />}
          <DangerousHTML
            html={translate("personality_heading", {
              deck_name: deck.name,
              personality: `<span>${personality.name}</span>`
            })}
            tag="h2"
          />
          {description && <p>{description.body}</p>}
        </div>,
        <div key="meaning" className={style.meaning}>
          <DangerousHTML html={translate("candidate_description_for_archetype_html")} tag="p" />
        </div>
      ]}
    </div>
  );
}

PersonalityArchetypeHeading.defaultProps = {assessment: null, deck: null};
PersonalityArchetypeHeading.propTypes = {
  assessment: PropTypes.shape({archetype: PropTypes.object}),
  deck: PropTypes.shape({name: PropTypes.string.isRequired}),
  followDeck: PropTypes.func.isRequired,
  getOption: PropTypes.func.isRequired,
  isReady: PropTypes.func.isRequired,
  options: PropTypes.shape({archetype: PropTypes.object}).isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityArchetypeHeading as Component};
export default withTraitify(PersonalityArchetypeHeading);
