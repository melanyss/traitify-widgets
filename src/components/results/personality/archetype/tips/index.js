import PropTypes from "prop-types";
import {useEffect, useState} from "react";
import {useDidMount, useDidUpdate} from "lib/helpers/hooks";
import {dig} from "lib/helpers/object";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

const tipTypes = [
  {disabledKey: "PersonalityTools", key: "tools", title: "Tools to Use"},
  {disabledKey: "PersonalityRoom", key: "room", title: "Room for Growth and Change"},
  {disabledKey: "PersonalitySettings", key: "settings", title: "Settings that Work for You"}
];

function PersonalityArchetypeTips(props) {
  const {assessment, getOption, isReady, translate, ui} = props;
  const details = dig(assessment, ["archetype", "details"]) || [];
  const [activeType, setActiveType] = useState(null);
  const [types, setTypes] = useState([]);
  const state = {activeType, types};

  useDidMount(() => { ui.trigger("PersonalityTips.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("PersonalityTips.updated", {props, state}); });
  useEffect(() => {
    if(details.length === 0) { return; }

    const disabledComponents = getOption("disabledComponents") || [];
    const activeTypes = tipTypes.filter((type) => {
      if(disabledComponents.includes(type.disabledKey)) { return false; }

      return details.find(({title}) => (title === type.title));
    });

    setTypes(activeTypes);
    setActiveType(activeTypes[0]);
  }, [details]);

  const disabledComponents = getOption("disabledComponents") || [];
  if(disabledComponents.includes("PersonalityTips")) { return null; }
  if(!isReady("results")) { return null; }
  if(!activeType) { return null; }

  const onChange = ({target: {value}}) => setActiveType(types.find((type) => type.key === value));
  const tips = details.filter(({title}) => (title === activeType.title)).map(({body}) => body);
  const typeStyle = {width: `${100.0 / types.length}%`};

  return (
    <div className={style.container}>
      <ul className={style.tabs}>
        {types.map((type) => (
          <li key={type.key} className={activeType.key === type.key ? style.active : ""} style={typeStyle}>
            <button onClick={() => setActiveType(type)} type="button">
              <div className={style.name}>{translate(`tip_type_for_${type.key}`)}</div>
            </button>
          </li>
        ))}
      </ul>
      <div className={style.tab}>
        <div className={style.formSelect}>
          <select onChange={onChange} value={activeType.key}>
            {types.map(({key}) => (
              <option key={key} value={key}>{translate(`tip_type_for_${key}`)}</option>
            ))}
          </select>
        </div>
        <ul className={style.list}>
          {tips.map((tip) => (<li key={tip}>{tip}</li>))}
        </ul>
      </div>
    </div>
  );
}

PersonalityArchetypeTips.defaultProps = {assessment: null};
PersonalityArchetypeTips.propTypes = {
  assessment: PropTypes.shape({archetype: PropTypes.object}),
  getOption: PropTypes.func.isRequired,
  isReady: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityArchetypeTips as Component};
export default withTraitify(PersonalityArchetypeTips);
