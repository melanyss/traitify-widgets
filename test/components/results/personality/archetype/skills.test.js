import {Component} from "components/results/personality/archetype/skills";
import ComponentHandler from "support/component-handler";
import _assessment from "support/json/assessment/dimension-based.json";

jest.mock("lib/with-traitify", () => ((value) => value));

const details = [];

[
  {body: "WFH Tip", name: "Working From Home Tips"},
  {body: "DWS Tip", name: "Dealing With Stress"},
  {body: "C Tip", name: "Communication Tips"},
  {body: "T Tip", name: "Teamwork"},
  {body: "SM Tip", name: "Self Motivation"}
].forEach((type) => {
  Array.from(Array(5)).forEach((_, i) => {
    details.push({
      body: `${type.body} - ${i}`,
      title: `Success Skills - ${type.name}`
    });
  });
});

const assessment = {
  ..._assessment,
  archetype: {
    ..._assessment.archetype,
    details: [
      ..._assessment.archetype.details,
      ...details
    ]
  }
};

describe("PersonalityArchetypeSkills", () => {
  let props;

  beforeEach(() => {
    props = {
      assessment,
      getOption: jest.fn().mockName("getOption"),
      isReady: jest.fn().mockName("isReady").mockReturnValue(true),
      options: {},
      translate: jest.fn().mockName("translate").mockImplementation((value, options = {}) => `${value}, ${options}`),
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalitySkills.initialized",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      props.ui.trigger.mockClear();
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalitySkills.updated",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
    });
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with available types", () => {
    props.assessment = {
      ...assessment,
      archetype: {
        ...assessment.archetype,
        details: assessment.archetype.details.filter((detail) => (
          detail.title !== "Success Skills - Working From Home Tips"
        ))
      }
    };
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with enabled types", () => {
    props.getOption.mockReturnValue(["Working From Home Tips"]);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders new type from clicking button", () => {
    const component = new ComponentHandler(<Component {...props} />);
    component.act(() => component.instance.findAllByType("button")[1].props.onClick());

    expect(component.tree).toMatchSnapshot();
  });

  it("renders new type from selecting option", () => {
    const component = new ComponentHandler(<Component {...props} />);
    const select = component.instance.findByType("select");
    const option = component.instance.findAllByType("option")[1];
    component.act(() => select.props.onChange({target: option.props}));

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if disabled", () => {
    props.getOption.mockReturnValue(["PersonalitySkills"]);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if results not ready", () => {
    props.assessment = null;
    props.isReady.mockImplementation((value) => value !== "results");
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if no archetype", () => {
    props.assessment = {...props.assessment, archetype: null};
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if no tips", () => {
    props.assessment = {
      ...props.assessment,
      archetype: {
        ...props.assessment.archetype,
        details: [] // TODO: Leave in other section details
      }
    };
  });
});
