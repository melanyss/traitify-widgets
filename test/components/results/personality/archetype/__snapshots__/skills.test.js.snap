// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityArchetypeSkills renders component 1`] = `
<div>
  <ul>
    <div>
      featured_skill, [object Object]
    </div>
    <li>
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_working_from_home, [object Object]"
          src="https://cdn.traitify.com/images/big5_home.png"
        />
        <div>
          skill_name_for_working_from_home, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_dealing_with_stress, [object Object]"
          src="https://cdn.traitify.com/images/big5_stress.png"
        />
        <div>
          skill_name_for_dealing_with_stress, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_communication, [object Object]"
          src="https://cdn.traitify.com/images/big5_communication.png"
        />
        <div>
          skill_name_for_communication, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_teamwork, [object Object]"
          src="https://cdn.traitify.com/images/big5_teamwork.png"
        />
        <div>
          skill_name_for_teamwork, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_self_motivation, [object Object]"
          src="https://cdn.traitify.com/images/big5_motivation.png"
        />
        <div>
          skill_name_for_self_motivation, [object Object]
        </div>
      </button>
    </li>
  </ul>
  <div>
    <div>
      <select
        onChange={[Function]}
        value="working_from_home"
      >
        <option
          value="working_from_home"
        >
          skill_name_for_working_from_home, [object Object]
        </option>
        <option
          value="dealing_with_stress"
        >
          skill_name_for_dealing_with_stress, [object Object]
        </option>
        <option
          value="communication"
        >
          skill_name_for_communication, [object Object]
        </option>
        <option
          value="teamwork"
        >
          skill_name_for_teamwork, [object Object]
        </option>
        <option
          value="self_motivation"
        >
          skill_name_for_self_motivation, [object Object]
        </option>
      </select>
    </div>
    <h3>
      skill_heading_for_working_from_home, [object Object]
    </h3>
    <ul>
      <li>
        WFH Tip - 0
      </li>
      <li>
        WFH Tip - 1
      </li>
      <li>
        WFH Tip - 2
      </li>
      <li>
        WFH Tip - 3
      </li>
      <li>
        WFH Tip - 4
      </li>
    </ul>
  </div>
</div>
`;

exports[`PersonalityArchetypeSkills renders component with available types 1`] = `
<div>
  <ul>
    <div>
      featured_skill, [object Object]
    </div>
    <li>
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_dealing_with_stress, [object Object]"
          src="https://cdn.traitify.com/images/big5_stress.png"
        />
        <div>
          skill_name_for_dealing_with_stress, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_communication, [object Object]"
          src="https://cdn.traitify.com/images/big5_communication.png"
        />
        <div>
          skill_name_for_communication, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_teamwork, [object Object]"
          src="https://cdn.traitify.com/images/big5_teamwork.png"
        />
        <div>
          skill_name_for_teamwork, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_self_motivation, [object Object]"
          src="https://cdn.traitify.com/images/big5_motivation.png"
        />
        <div>
          skill_name_for_self_motivation, [object Object]
        </div>
      </button>
    </li>
  </ul>
  <div>
    <div>
      <select
        onChange={[Function]}
        value="dealing_with_stress"
      >
        <option
          value="dealing_with_stress"
        >
          skill_name_for_dealing_with_stress, [object Object]
        </option>
        <option
          value="communication"
        >
          skill_name_for_communication, [object Object]
        </option>
        <option
          value="teamwork"
        >
          skill_name_for_teamwork, [object Object]
        </option>
        <option
          value="self_motivation"
        >
          skill_name_for_self_motivation, [object Object]
        </option>
      </select>
    </div>
    <h3>
      skill_heading_for_dealing_with_stress, [object Object]
    </h3>
    <ul>
      <li>
        DWS Tip - 0
      </li>
      <li>
        DWS Tip - 1
      </li>
      <li>
        DWS Tip - 2
      </li>
      <li>
        DWS Tip - 3
      </li>
      <li>
        DWS Tip - 4
      </li>
    </ul>
  </div>
</div>
`;

exports[`PersonalityArchetypeSkills renders component with enabled types 1`] = `
<div>
  <ul>
    <div>
      featured_skill, [object Object]
    </div>
    <li>
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_dealing_with_stress, [object Object]"
          src="https://cdn.traitify.com/images/big5_stress.png"
        />
        <div>
          skill_name_for_dealing_with_stress, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_communication, [object Object]"
          src="https://cdn.traitify.com/images/big5_communication.png"
        />
        <div>
          skill_name_for_communication, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_teamwork, [object Object]"
          src="https://cdn.traitify.com/images/big5_teamwork.png"
        />
        <div>
          skill_name_for_teamwork, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_self_motivation, [object Object]"
          src="https://cdn.traitify.com/images/big5_motivation.png"
        />
        <div>
          skill_name_for_self_motivation, [object Object]
        </div>
      </button>
    </li>
  </ul>
  <div>
    <div>
      <select
        onChange={[Function]}
        value="dealing_with_stress"
      >
        <option
          value="dealing_with_stress"
        >
          skill_name_for_dealing_with_stress, [object Object]
        </option>
        <option
          value="communication"
        >
          skill_name_for_communication, [object Object]
        </option>
        <option
          value="teamwork"
        >
          skill_name_for_teamwork, [object Object]
        </option>
        <option
          value="self_motivation"
        >
          skill_name_for_self_motivation, [object Object]
        </option>
      </select>
    </div>
    <h3>
      skill_heading_for_dealing_with_stress, [object Object]
    </h3>
    <ul>
      <li>
        DWS Tip - 0
      </li>
      <li>
        DWS Tip - 1
      </li>
      <li>
        DWS Tip - 2
      </li>
      <li>
        DWS Tip - 3
      </li>
      <li>
        DWS Tip - 4
      </li>
    </ul>
  </div>
</div>
`;

exports[`PersonalityArchetypeSkills renders new type from clicking button 1`] = `
<div>
  <ul>
    <div>
      featured_skill, [object Object]
    </div>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_working_from_home, [object Object]"
          src="https://cdn.traitify.com/images/big5_home.png"
        />
        <div>
          skill_name_for_working_from_home, [object Object]
        </div>
      </button>
    </li>
    <li>
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_dealing_with_stress, [object Object]"
          src="https://cdn.traitify.com/images/big5_stress.png"
        />
        <div>
          skill_name_for_dealing_with_stress, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_communication, [object Object]"
          src="https://cdn.traitify.com/images/big5_communication.png"
        />
        <div>
          skill_name_for_communication, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_teamwork, [object Object]"
          src="https://cdn.traitify.com/images/big5_teamwork.png"
        />
        <div>
          skill_name_for_teamwork, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_self_motivation, [object Object]"
          src="https://cdn.traitify.com/images/big5_motivation.png"
        />
        <div>
          skill_name_for_self_motivation, [object Object]
        </div>
      </button>
    </li>
  </ul>
  <div>
    <div>
      <select
        onChange={[Function]}
        value="dealing_with_stress"
      >
        <option
          value="working_from_home"
        >
          skill_name_for_working_from_home, [object Object]
        </option>
        <option
          value="dealing_with_stress"
        >
          skill_name_for_dealing_with_stress, [object Object]
        </option>
        <option
          value="communication"
        >
          skill_name_for_communication, [object Object]
        </option>
        <option
          value="teamwork"
        >
          skill_name_for_teamwork, [object Object]
        </option>
        <option
          value="self_motivation"
        >
          skill_name_for_self_motivation, [object Object]
        </option>
      </select>
    </div>
    <h3>
      skill_heading_for_dealing_with_stress, [object Object]
    </h3>
    <ul>
      <li>
        DWS Tip - 0
      </li>
      <li>
        DWS Tip - 1
      </li>
      <li>
        DWS Tip - 2
      </li>
      <li>
        DWS Tip - 3
      </li>
      <li>
        DWS Tip - 4
      </li>
    </ul>
  </div>
</div>
`;

exports[`PersonalityArchetypeSkills renders new type from selecting option 1`] = `
<div>
  <ul>
    <div>
      featured_skill, [object Object]
    </div>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_working_from_home, [object Object]"
          src="https://cdn.traitify.com/images/big5_home.png"
        />
        <div>
          skill_name_for_working_from_home, [object Object]
        </div>
      </button>
    </li>
    <li>
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_dealing_with_stress, [object Object]"
          src="https://cdn.traitify.com/images/big5_stress.png"
        />
        <div>
          skill_name_for_dealing_with_stress, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_communication, [object Object]"
          src="https://cdn.traitify.com/images/big5_communication.png"
        />
        <div>
          skill_name_for_communication, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_teamwork, [object Object]"
          src="https://cdn.traitify.com/images/big5_teamwork.png"
        />
        <div>
          skill_name_for_teamwork, [object Object]
        </div>
      </button>
    </li>
    <li
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="skill_name_for_self_motivation, [object Object]"
          src="https://cdn.traitify.com/images/big5_motivation.png"
        />
        <div>
          skill_name_for_self_motivation, [object Object]
        </div>
      </button>
    </li>
  </ul>
  <div>
    <div>
      <select
        onChange={[Function]}
        value="dealing_with_stress"
      >
        <option
          value="working_from_home"
        >
          skill_name_for_working_from_home, [object Object]
        </option>
        <option
          value="dealing_with_stress"
        >
          skill_name_for_dealing_with_stress, [object Object]
        </option>
        <option
          value="communication"
        >
          skill_name_for_communication, [object Object]
        </option>
        <option
          value="teamwork"
        >
          skill_name_for_teamwork, [object Object]
        </option>
        <option
          value="self_motivation"
        >
          skill_name_for_self_motivation, [object Object]
        </option>
      </select>
    </div>
    <h3>
      skill_heading_for_dealing_with_stress, [object Object]
    </h3>
    <ul>
      <li>
        DWS Tip - 0
      </li>
      <li>
        DWS Tip - 1
      </li>
      <li>
        DWS Tip - 2
      </li>
      <li>
        DWS Tip - 3
      </li>
      <li>
        DWS Tip - 4
      </li>
    </ul>
  </div>
</div>
`;

exports[`PersonalityArchetypeSkills renders nothing if disabled 1`] = `null`;

exports[`PersonalityArchetypeSkills renders nothing if no archetype 1`] = `null`;

exports[`PersonalityArchetypeSkills renders nothing if results not ready 1`] = `null`;
